src/realm/group_writer.cpp

```diff
@@ -731,7 +1023,7 @@ GroupWriter::FreeListElement GroupWriter::search_free_space_in_part_of_freelist(
     while (it != m_size_map.end()) {
         // Accept either a perfect match or a block that is twice the size. Tests have shown
         // that this is a good strategy.
-        if (it->first == size || it->first >= 2 * size) {
+        if (it->first >= size || it->first >= 2 * size) {
```

Comment:
Please don't change the allocator strategy in the middle of this massive commit. I think this deserves to be its own commit.

(also, the second half is redundant - it is only true if the first part is)

Reply:
oh, that is a bug. thx.